### PR TITLE
verify transaction status before confirming

### PR DESCRIPTION
**Pull request by blackdevelopa**
## Explanation 

If a transaction fails just after it's added to a block, we initially default to marking the transaction as confirmed (success). Here, we are adding an extra check that verifies status before confirming.

## References
https://github.com/MetaMask/metamask-mobile/issues/6834
https://github.com/MetaMask/metamask-mobile/pull/7350

### `@metamask/transaction-controller`

- **ADDED**: Call `checkTxReceiptStatusIsFailed` after transaction is added to a block 

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/TransactionController.ts [hunk 1]
```diff
@@ -982,9 +982,19 @@ export class TransactionController extends BaseController<
...
         if (txObj?.blockNumber) {
-          meta.status = TransactionStatus.confirmed;
-          this.hub.emit(`${meta.id}:confirmed`, meta);
-          return [meta, true];
+          // transactions can be added to a block and still fail, so we need to check the transaction status before emitting the confirmed event
+          const verifyTransactionStatus =
+            await this.checkTxReceiptStatusIsFailed(transactionHash);
+          if (verifyTransactionStatus) {
+            const error = new Error(
+              'Transaction failed. Error encountered during contract execution [out of gas]',
```
Comment (blackdevelopa) [inline packages/transaction-controller/src/TransactionController.ts:990]: Is this error message sufficient or valid? Here's a link to a failed [transaction](https://goerli.etherscan.io/tx/0xb5e68e55fc2a199c169e173fcf7644b5e6273d3d0a2492823f2093e4203b2670)

File: packages/transaction-controller/src/TransactionController.ts [hunk 2]
```diff
@@ -982,9 +982,19 @@ export class TransactionController extends BaseController<
 
         /* istanbul ignore next */
         if (txObj?.blockNumber) {
-          meta.status = TransactionStatus.confirmed;
-          this.hub.emit(`${meta.id}:confirmed`, meta);
-          return [meta, true];
+          // transactions can be added to a block and still fail, so we need to check the transaction status before emitting the confirmed event
+          const verifyTransactionStatus =
+            await this.checkTxReceiptStatusIsFailed(transactionHash);
```
Comment (jpuri) [inline packages/transaction-controller/src/TransactionController.ts:987]: Variable name `verifyTransactionStatus` is not intuitive.

**Review by jpuri — APPROVED** [review]
Changes looks good, I added a small nit.

Additionally we should ensure this change is merged to main and also test covered.

File: packages/transaction-controller/src/TransactionController.ts (hunk 2 above)
Comment (blackdevelopa) [reply packages/transaction-controller/src/TransactionController.ts:987]: I updated this to `checkIfTxStatusIsFailed`. Does this work?

File: packages/transaction-controller/src/TransactionController.ts (hunk 1 above)
Comment (blackdevelopa) [reply packages/transaction-controller/src/TransactionController.ts]: I updated this error message to match the message in line [957](https://github.com/MetaMask/core/blob/7163988613a645bd94a8ee456c3660cafb305b79/packages/transaction-controller/src/TransactionController.ts#L957)

File: packages/transaction-controller/src/TransactionController.ts (hunk 2 above)
Comment (blackdevelopa) [reply packages/transaction-controller/src/TransactionController.ts]: Updated. https://github.com/MetaMask/core/pull/1773/commits/0f49dd6f37110c2f6755d07cc50a9de47015b3af

**Pull request merged**